### PR TITLE
Add the Mean Squared Logarithmic Error (MSLE) loss function

### PR DESCRIPTION
**Pull request by AtomicVar**
## Proposed changes

Mean Squared Logarithmic Error (MSLE) is a loss function used in regression problems. Some ML frameworks/libraries contain this loss function, including:

- [TensorFlow (tf.keras.losses.MeanSquaredLogarithmicError)](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MeanSquaredLogarithmicError)
- [torchmetrics (MeanSquaredLogError)](https://lightning.ai/docs/torchmetrics/stable/regression/mean_squared_log_error.html)
- [scikit-learn (mean_squared_log_error)](https://scikit-learn.org/stable/modules/model_evaluation.html#mean-squared-log-error)

This PR adds a `msle_loss` function to `nn.losses` module.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Review by Stealeristaken — CHANGES_REQUESTED** [review]
Maybe adding 
```python
 if (targets < 0).any() or (inputs < 0).any():
        raise ValueError(
            "Mean Squared Logarithmic Error cannot be used when "
            "targets contain negative values."
        )
```

 could be nice and 
```python
loss=mse_loss(mx.log1p(targets), mx.log1p(inputs))
```


can be more optimized.

**Comment by awni** [comment]
If you do something like `if (targets < 0).any()` it will eval the graph, so that is not a good idea. It won't break gradient transforms, but it could be very slow.

I think the implementation is fine the way you have it, that said I don't think we should include this in `nn.losses` yet. It is pretty niche and also reasonably easy to implement so I don't see a strong need to include it here. We can revisit if that  changes in the future. In the meantime, I will close this.

**Comment by Stealeristaken** [comment]
> If you do something like `if (targets < 0).any()` it will eval the graph, so that is not a good idea. It won't break gradient transforms, but it could be very slow.
> 
> I think the implementation is fine the way you have it, that said I don't think we should include this in `nn.losses` yet. It is pretty niche and also reasonably easy to implement so I don't see a strong need to include it here. We can revisit if that changes in the future. In the meantime, I will close this.

I know its a bit late response sorry but what do you mean by `it will eval the graph` and why is it a bad idea? Couldn't figure out

**Comment by awni** [comment]
You can read about how lazy evaluation works in MLX in the [docs](https://ml-explore.github.io/mlx/build/html/usage/lazy_evaluation.html). I think it should answer your question.
